core/net/ipv6/uip-nd6.c

```diff
@@ -662,7 +662,8 @@ rs_input(void)
         }
         if(memcmp(&nd6_opt_llao[UIP_ND6_OPT_DATA_OFFSET],
             lladdr, UIP_LLADDR_LEN) != 0) {
-          uip_ds6_nbr_t nbr_data = *nbr;
+          uip_ds6_nbr_t nbr_data;
+          nbr_data = *nbr;
```

Comment:
> On my machine, running latest git cc65:
> [...]
> Am I using an unsupported cc65? Which one should be used with Contiki?

You're using the one and only cc65 suitable for Contiki. It's the one used by Travis too: https://github.com/contiki-os/contiki/blob/master/.travis.yml#L74

Thanks for isolating the issue with a small test program. I'll post it on the cc65 mailing list...